src/main/java/com/google/error_analyzer/backend/StackTrace.java

```diff
@@ -0,0 +1,144 @@
+/**Copyright 2019 Google LLC
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    https://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.*/
+
+package com.google.error_analyzer.backend;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableList.Builder;
+import com.google.error_analyzer.backend.LogDaoHelper;
+import com.google.error_analyzer.data.constant.LogFields;
+import com.google.error_analyzer.data.constant.StackTraceFormat;
+import java.io.IOException;
+import java.lang.*;
+import java.util.*;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.LogManager;
+import org.elasticsearch.action.search.SearchRequest;
+import org.elasticsearch.index.query.RangeQueryBuilder;
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+import org.elasticsearch.search.SearchHit;
+import org.elasticsearch.search.SearchHits;
+/**
+* Find the stack trace following an error. The algorithm fetches documents from
+* the index and see if they fit they stacktrace format.
+*/
+public class StackTrace {
+    private Integer BATCH_SIZE = 100;
+    private Integer ALLOWED_MESSAGES = 5;
+    private static final Logger logger = LogManager.getLogger(StackTrace.class);
+    public DaoInterface logDao = new LogDao();
+
+    // Returns logText of the documents which form the stack trace
+    // following an error
+    public ImmutableList < String > findStack(Integer errorLogLineNumber,
+    String fileName) throws IOException {
+        SearchRequest searchRequest = createSearchRequest(fileName,
+        errorLogLineNumber);
+        logger.info("Finding stack for ".concat(errorLogLineNumber.toString()));
+        ImmutableList < SearchHit > rangeHits = logDao.getHitsFromIndex(searchRequest);
+        Builder < String > stackLogLinesBuilder = ImmutableList.< String > builder();
+        Integer startOfStack = findStartOfStack(rangeHits);
+        if (startOfStack == -1) {
+            return stackLogLinesBuilder.build();
+        } else {
+            //add message brfore stack starts to stackLogLines
+            logger.info("Adding ".concat(startOfStack.toString())
+            .concat(" messages to stack list"));
+            stackLogLinesBuilder.addAll(extractLogTextFormHits(rangeHits, 0, startOfStack));
+        }
+        ImmutableList < String > stackList = iterateHitsForFindingStack(rangeHits, startOfStack);
```

Comment:
Can we move 57-60 inside while loop.

Reply:
Improves readability as the first case is different from the case inside while. Also it needs a different condition to check if end of stack trace was found .